vcs/git/clone_update.go

```diff
@@ -93,10 +93,11 @@ func (r *Repository) UpdateEverything(opt vcs.RemoteOpts) error {
 	}
 
 	if err := rm.Fetch([]string{"+refs/*:refs/*"}, &opts, ""); err != nil {
-		return err
+		return vcs.UpdateResult{}, err
 	}
 
-	return nil
+	// TODO: Calculate value of vcs.UpdateResult.
```

Comment:
What are the semantics of an empty `vcs.UpdateResult`? If it means "nothing changed", then this is technically incorrect, right? If a client happens to be using the `git` implementation with the expectation that if `UpdateResult` is empty, then nothing is changed, then the client might proceed incorrectly. Wouldn't it be better to add a `panic` here, so that if any client tries to call this, it fails? Or alternatively, have an option field in `vcs.RemoteOpts` to request that a result is returned.